api/src/test/java/org/openmrs/api/EncounterServiceTest.java

```diff
@@ -1048,6 +1099,22 @@ public void unvoidEncounter_shouldCascadeUnvoidToOrders() {
 		assertNull(order.getVoidReason());
 	}
 	
+	/**
+	 * @see EncounterService#unvoidEncounter(Encounter)
+	 */
+	@Test
+	public void unvoidEncounter_shouldCascadeUnvoidToDiagnoses() {
+		EncounterService encounterService = Context.getEncounterService();
+
+		// get a voided encounter that has some voided Diagnoses
+		Encounter encounter = encounterService.getEncounter(2);
+		encounterService.unvoidEncounter(encounter);
+
+		Diagnosis diagnosis = Context.getDiagnosisService().getDiagnosis(2);
```

Comment:
Can you assert that the diagnosis was voided before the un-voiding of the encounter?